### ISSUE TITLE
(5.x) preventDefaultTouchmoveEvent (Mobile Safari?)

### ISSUE THREAD

**Issue by bhj** [dogfessional/react-swipeable#127]
Hi and thanks for all your work.

I can't seem to prevent vertical scrolling during a left/right swipe using `preventDefaultTouchmoveEvent` with 5.x where 4.x works fine (iOS 12.1.4, React 16.8.3). I'm using it with `onSwipedLeft` and `onSwipedRight` and everything else default. Seeing same behavior in [the example](http://dogfessional.github.io/react-swipeable/) actually; any ideas or anyone else experiencing this?

**Comment by hartzis** [comment]
Thank you for posting this issue. I'll try to allocate some time next week to dig into this, but off the top of my head it could be related to the checks here:

https://github.com/dogfessional/react-swipeable/blob/e6f79e9cbf582f80ab13f8309f1faf8af8a6a7de/src/index.js#L95-L100

This was copied from v4. It only allows preventDefault to be called when there is a handler for the direction the user is swiping.

### Fixes
- could add `onSwiping={() => {}}` prop to your `<Swipeable>` usage along with `preventDefaultTouchmoveEvent` and it should work.

- could also try calling `e.preventDefault()` on the event within your code to.

```
handleSwipedLeft = ({event}) => {
  event.preventDefault();
  console.log('swipedLeft')
}
...
onSwipedLeft={this.handleSwipedLeft}
```

We may need to explore adding a new prop to **always** preventDefault?

...

> (btw the link to migrations.md in the readme and the "example migration" link in migrations.md are currently borked)
Thank you for pointing this out!

**Comment by JiiB** [comment]
I currently have the same issue 😞

**Comment by JiiB** [comment]
@bhj did you find a workaround?

**Comment by JiiB** [comment]
@bhj @hartzis I'm going to create a PR to fix this issue

**Comment by hartzis** [comment]
@JiiB what are your thoughts on an approach or we can just discuss when the PR is open.

**Comment by hartzis** [comment]
probable related ios/safari issue - https://bugs.webkit.org/show_bug.cgi?id=182521

**Comment by hartzis** [comment]
Essentially the root issue - https://github.com/facebook/react/issues/2043

**Comment by hartzis** [comment]
@JiiB I think i got a better solution after a lot more digging in today.

Please let me know what you think of #131 

Also published it as `react-swipeable@5.1.0-alpha.1`

**Comment by JiiB** [comment]
@hartzis @bhj `react-swipeable@5.1.0-alpha.1` works for me. you can now call `event.preventDefault()` inside of your `onSwiping` handler 👍

**Comment by bhj** [comment]
@hartzis Alpha is working over here too. Thanks for digging into this!

**Comment by hartzis** [comment]
Thank you @JiiB and @bhj !

I genuinely appreciate you both for testing and verifying the alpha version.

**Comment by hartzis** [comment]
- found small bug with #131
- created fix #132 
- published fix as `react-swipeable@5.1.0-alpha.2`

Still on target to release `5.1.0` this week.
